src/plugins/expression_error/public/expression_renderers/debug_renderer.tsx

```diff
@@ -36,21 +33,20 @@ const strings = {
 };
 
 export const getDebugRenderer =
-  (theme$: Observable<CoreTheme> = defaultTheme$) =>
-  (): ExpressionRenderDefinition<any> => ({
+  (theme: ThemeServiceSetup) => (): ExpressionRenderDefinition<any> => ({
     name: 'debug',
     displayName: strings.getDisplayName(),
     help: strings.getHelpDescription(),
     reuseDomNode: true,
     render(domNode, config, handlers) {
       handlers.onDestroy(() => unmountComponentAtNode(domNode));
       render(
-        <KibanaThemeProvider theme$={theme$}>
+        <KibanaThemeProvider theme={theme}>
```

Comment:
Unfortunately, it's not simple to use `KibanaRenderContextProvider` here. Due to `any` typing in Canvas, the core object being passed is `CoreStart`. In other calling contexts, we can call `core.getStartServices` with the async render function, but breaks in Canvas.

Reply:
Thinking more on this, since we can't easily use `KibanaRenderContextProvider`, we should probably wrap this in `KibanaErrorBoundary`. That should be trivial since it only requires an analytics service dependency (for telemetry reported when error happens)